ferrypick.py

```diff
@@ -97,15 +99,87 @@ def get_patch_content(link):
     return (content, original_name)
 
 
+def handle_reject(filename):
+    """If the .rej file given in `filename` is "simple", run rmpdev-bumpspec
+
+    Simple means roughly that only Release lines are touched and
+    %changelog lines are added.
+
+    Removes the reject file if successful.
+    """
+    changelog = None
+    path = Path(filename)
+    author = None
+    with path.open() as f:
+        for line in f:
+            # Find first hunk header
+            if line.startswith('@'):
+                break
+        for line in f:
+            marker = line[:1]
+            print(line.rstrip())
+            if marker == '@':
+                # Hunk header
+                continue
+            elif marker == ' ':
+                # Context
+                if line.strip() == '%changelog':
+                    changelog = []
+                continue
+            elif marker in ('+', '-'):
+                if changelog is not None:
+                    if marker == '-':
+                        # Removing existing changelog - bad
+                        return
+                    if match := re.match(
+                        r'\*\s+(\S+\s+){4}(?P<author>[^>]+>)',
+                        line[1:]
+                    ):
+                        author = match['author']
+                    else:
+                        changelog.append(line[1:])
+                elif line[1:].startswith('Release:'):
+                    continue
+                else:
+                    # Adding/removing something else - bad
+                    return
+            else:
+                # Unknown line - bad
+                return
+    if author is None:
+        print('No author found in reject')
+        return
+    print(f'Rejects in {filename} look harmless')
+    execute(
+        'rpmdev-bumpspec',
+        '-u', author,
+        '-c', ''.join(changelog).strip(),
+        path.with_suffix(''),
+        check=True,
+    )
+    path.unlink()
+
+
 def apply_patch(filename):
-    cmd = f"git am --committer-date-is-author-date --reject {filename}"
-    print(f"$ {cmd}")
-    exitcode = execute(cmd)
+    args = [
+        "git", "am", "--committer-date-is-author-date", "--reject", filename,
+    ]
+    exitcode = execute(*args).returncode
     if exitcode:
         print(file=sys.stderr)
-        print(f"{cmd} failed with exit code {exitcode}", file=sys.stderr)
+        print(f"git am failed with exit code {exitcode}", file=sys.stderr)
         print(f"Patch stored as: {filename}", file=sys.stderr)
-        sys.exit(exitcode)
+
+        for spec_rej in Path().glob(f'**/*.spec.rej'):
```

Comment:
I wonder whether this can be a leftover rej file from previous invocation?

Reply:
It can.
I guess it would be better to scan for `.rej` files before `am`, and disable the bumping if any are found